plugins/woocommerce/src/Internal/FraudProtection/SessionClearanceManager.php

```diff
@@ -0,0 +1,223 @@
+<?php
+/**
+ * SessionClearanceManager class file.
+ */
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\FraudProtection;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Manages session clearance state for fraud protection.
+ *
+ * This class handles the session status tracking for fraud protection decisions,
+ * managing three possible states: pending, allowed, and blocked. It integrates
+ * with WooCommerce sessions and uses the FraudProtectionController logging helper
+ * to maintain consistent audit logs.
+ *
+ * @since 10.5.0
+ * @internal This class is part of the internal API and is subject to change without notice.
+ */
+class SessionClearanceManager {
+
+	/**
+	 * Session key for storing clearance status.
+	 */
+	private const SESSION_KEY = '_fraud_protection_clearance_status';
+
+	/**
+	 * Session status: pending clearance.
+	 */
+	public const STATUS_PENDING = 'pending';
+
+	/**
+	 * Session status: allowed.
+	 */
+	public const STATUS_ALLOWED = 'allowed';
+
+	/**
+	 * Session status: blocked.
+	 */
+	public const STATUS_BLOCKED = 'blocked';
+
+	/**
+	 * Default session status.
+	 */
+	public const DEFAULT_STATUS = self::STATUS_ALLOWED;
+
+	/**
+	 * Check if the current session is allowed.
+	 *
+	 * @return bool True if session is allowed, false otherwise.
+	 */
+	public function is_session_allowed(): bool {
+		$status = $this->get_session_status();
+		return self::STATUS_ALLOWED === $status;
+	}
+
+	/**
+	 * Check if the current session is blocked.
+	 *
+	 * @return bool True if session is blocked, false otherwise.
+	 */
+	public function is_session_blocked(): bool {
+		$status = $this->get_session_status();
+		return self::STATUS_BLOCKED === $status;
+	}
+
```

Comment:
I'm not sure if it'd be useful in the future, but perhaps for consistency we could add an `is_session_challenged` method as well?

Reply:
Let's add it later when we work on challenge features if it's necessary. It might be better for certain actions to explicitly check if they are allowed or blocked. It's important to know though that `NOT blocked !== allowed`